lib/static/new-ui/features/suites/components/TreeActionsToolbar/index.tsx

```diff
@@ -0,0 +1,189 @@
+import {
+    SquareCheck,
+    ChevronsExpandVertical,
+    ChevronsCollapseVertical,
+    SquareDashed,
+    Square,
+    CircleInfo,
+    Play,
+    Check,
+    ArrowUturnCcwLeft
+} from '@gravity-ui/icons';
+import React, {ReactNode, useMemo} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import styles from './index.module.css';
+import {
+    acceptOpened,
+    collapseAll,
+    deselectAll,
+    expandAll,
+    retrySuite,
+    selectAll, staticAccepterStageScreenshot, staticAccepterUnstageScreenshot,
+    undoAcceptImages
+} from '@/static/modules/actions';
+import {ImageEntity, State} from '@/static/new-ui/types/store';
+import {CHECKED, INDETERMINATE} from '@/constants/checked-statuses';
+import {IconButton} from '@/static/new-ui/components/IconButton';
+import {
+    getCheckedTests,
+    getSelectedImages,
+    getVisibleBrowserIds,
+    getVisibleImages
+} from '@/static/modules/selectors/tree';
+import {getIsGui, getIsInitialized, getIsStaticImageAccepterEnabled} from '@/static/new-ui/store/selectors';
+import {isAcceptable, isScreenRevertable} from '@/static/modules/utils';
+import {EditScreensFeature, RunTestsFeature} from '@/constants';
+
+interface TreeActionsToolbarProps {
+    onHighlightCurrentTest?: () => void;
+}
+
+export function TreeActionsToolbar(props: TreeActionsToolbarProps): ReactNode {
+    const dispatch = useDispatch();
+
+    const rootSuiteIds = useSelector((state: State) => state.tree.suites.allRootIds);
+    const suitesStateById = useSelector((state: State) => state.tree.suites.stateById);
+    const browsersStateById = useSelector((state: State) => state.tree.browsers.stateById);
+    const browsersById = useSelector((state: State) => state.tree.browsers.byId);
+    const selectedTests = useSelector(getCheckedTests);
+    const visibleBrowserIds: string[] = useSelector(getVisibleBrowserIds);
+    const isInitialized = useSelector(getIsInitialized);
+
+    const isRunTestsAvailable = useSelector((state: State) => state.app.availableFeatures)
+        .find(feature => feature.name === RunTestsFeature.name);
+    const isRunning = useSelector((state: State) => state.running);
+
+    const isEditScreensAvailable = useSelector((state: State) => state.app.availableFeatures)
+        .find(feature => feature.name === EditScreensFeature.name);
+
+    const isSelectedAll = useMemo(() => {
+        for (const suiteId of rootSuiteIds) {
+            if (suitesStateById[suiteId].checkStatus !== CHECKED) {
+                return false;
+            }
+        }
+
+        return true;
+    }, [suitesStateById, rootSuiteIds]);
+
+    const isSelectedAtLeastOne = useMemo(() => {
+        for (const suiteId of rootSuiteIds) {
+            if (suitesStateById[suiteId].shouldBeShown && (suitesStateById[suiteId].checkStatus === CHECKED || suitesStateById[suiteId].checkStatus === INDETERMINATE)) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
Why its not `Array.some`?
```ts
return rootSuiteIds.some(suiteId => {
    const isShown = suitesStateById[suiteId].shouldBeShown;
    const isChecked = [CHECKED, INDETERMINATE].includes(suitesStateById[suiteId].checkStatus);

    return isShown && isChecked;
});
```

Reply:
Pretty sure it's a matter of taste, but okay, switched it to `some()`.